File: PAT/BLevel/1017/1017.cpp
// 1017 A除以B
// auguralpha
// 2019-02-25

#include <iostream>
#include <string>
using namespace std;

int main() {
    string a;
    int b;
    int q = 0;
    int r = 0;
    cin >> a >> b;
    int len_a = int(a.size());
    q = (a[0] - '0') / b;
    if (len_a == 1 || (q != 0 && len_a > 1)) {
        cout << q;
    }
    r = (a[0] - '0') % b;
    for(int i = 1; i < len_a; ++ i) {
        q = (a[i] - '0' + r * 10) / b;
        r = (a[i] - '0' + r * 10) % b;
        cout << q;
    }
    cout << " " << r << endl;
    return 0;
}

File: PAT/BLevel/1017/1017.go
// 1017 A除以B
// auguralpha
// 2020-03-26

package main

import "fmt"

func main() {
	var a string
	var b int
	var q int = 0
	var r int = 0

	_, _ = fmt.Scanf("%s %d", &a, &b)

	q = int(a[0] - '0') / b
	if len(a) == 1 || (q != 0 && len(a) > 1) {
		fmt.Printf("%d", q)
	}

	r = int(a[0] - '0') % b
	for i := 1; i < len(a); i ++ {
		q =  (int(a[i] - '0') + r * 10) / b
		r =  (int(a[i] - '0') + r * 10) % b
		fmt.Printf("%d", q)
	}
	fmt.Printf(" %d\n", r)
}
